lib/hcrypto/evp.c

```diff
@@ -969,7 +970,7 @@ EVP_CipherFinal_ex(EVP_CIPHER_CTX *ctx, void *out, int *outlen)
  * @param in in data to the operation.
  * @param size length of data.
  *
- * @return 1 on success.
+ * @return bytes encrypted on success, zero on failure.
```

Comment:
At line 979 below do_cipher() is going to return 1 for success and any other value (less than or equal to zero) is an error.  Therefore the return value from EVP_Cipher() must not be the return from do_cipher() and instead be either zero or number of bytes.   If we have nothing else, then return 'size' on success.

Reply:
The OpenSSL docs say:

```
EVP_CipherFinal_ex() returns 0 for a decryption failure or 1 for success.
```